### ISSUE TITLE
Java generator crashes for large grammars

### ISSUE THREAD

**Issue by nverwer** [orlandohill/waxeye#13]
The Java code generator initializes the states and edges in one method `makeAutomata`.
When the grammar has many rules, this method becomes so large, that the Java compiler gives an error "code too large". This is because the bytecode for a single method in Java cannot exceed 64kB.

A solution might be to divide the code of the `makeAutomata` method into smaller pieces, like one piece per non-terminal.
The `gen-make-automata` in java.scm generates this code, by mapping the function `gen-fa` on the list `automata`. Instead, it could call a method named with the  `camel-case-upper` of the non-terminal symbol, and later define all these methods using `gen-fa`.

I have never programmed Scheme, but I will see if I can get this working. If I do, I will make a pull request. If someone who knows Scheme (Orlando?) can do this that would be much appreciated.

Edit: I think I almost have a solution, which I will test and submit as a pull request tomorrow.

**Comment by nverwer** [comment]
I think I came some way towards a solution, but my lack of Scheme knowledge prevents me to get it right. My idea was to initialize the stage and edges for each automaton in a separate Java method. Since there are '!' and '&' automata, I wanted to assign a number to these, and that is where things go wrong.

Maybe someone can look at commit `a459e17e6681c8069771b8062f42dc618620d1d8` of `https://github.com/nverwer/waxeye.git` and correct my beginner's mistakes? That would be much appreciated.

**Comment by nverwer** [comment]
I managed to learn just enough about Scheme to get this working, and will create a pull request.
However, this still contains some suspicious code: I ended up creating my own `range` function, because I could not get `in-range` or `for/list` to work correctly.

Also, on my machine (Windows 8 64-bit) the generated exe has a size of 5.6 MB, whereas the 'official' binary I downloaded before was only 4.1 MB. Maybe this is because I compiled with Racket, not PLT?
